### PR TITLE
chore: turn on Typescript's noUnusedLocals

### PR DESCRIPTION
**Pull request by riccardo-forina**
I noticed quite a bit of unused code around the app. I enabled the relative tsconfig option to get an error when this happens, and fixed what I found.

### CODE REVIEW

File: app/ui/src/app/connections/common/configuration/configuration.service.ts [hunk 1]
```diff
@@ -45,7 +45,7 @@ export class ConnectionConfigurationService {
         break;
       default:
     }
-    const formModel = this.formFactory.createFormModel(
+    this.formFactory.createFormModel(
```
Comment (gashcrumb) [inline app/ui/src/app/connections/common/configuration/configuration.service.ts]: You can actually delete this block I think, the next line is the same thing but with a return

File: app/ui/src/app/core/providers/integration-support-provider.service.ts [hunk 2]
```diff
@@ -183,6 +183,7 @@ export class IntegrationSupportProviderService extends IntegrationSupportService
       .get();
   }
 
+  // @ts-ignore
```
Comment (gashcrumb) [inline app/ui/src/app/core/providers/integration-support-provider.service.ts]: Maybe on these can you stick a `TODO` line on?  I think actually we could remove these two functions

File: app/ui/src/app/platform/types/integration/integration.reducer.ts [hunk 3]
```diff
@@ -210,7 +210,7 @@ export function integrationReducer(
     case IntegrationActions.DELETE_INTEGRATION_FAIL: {
       const error = (action as IntegrationActions.IntegrationDeleteFail)
         .payload;
-      const restCollection = state.collection.filter(
+      state.collection.filter(
```
Comment (gashcrumb) [inline app/ui/src/app/platform/types/integration/integration.reducer.ts]: I think this block could be deleted altogether.

File: app/ui/src/app/settings/settings-root.component.ts [hunk 4]
```diff
@@ -1,11 +1,11 @@
 import { Component } from '@angular/core';
-import { ActivatedRoute, Router } from '@angular/router';
+// import { ActivatedRoute, Router } from '@angular/router';
```
Comment (gashcrumb) [inline app/ui/src/app/settings/settings-root.component.ts]: Should we just delete these lines?

File: app/ui/src/app/store/connector/connector.service.ts [hunk 5]
```diff
@@ -43,7 +43,7 @@ export class ConnectorService extends RESTService<Connector, Connectors> {
         // Try and clear any stale cookies, though we can't touch HttpOnly ones
         document.cookie.split(';').forEach(function(c) {
           if (c.startsWith('cred-')) {
-            const newCookie = c
+            c
```
Comment (gashcrumb) [inline app/ui/src/app/store/connector/connector.service.ts]: I think we can delete this whole split/forEach block starting with the comment on line 43 there

File: app/ui/src/app/store/integration/integration.service.ts [hunk 6]
```diff
@@ -91,6 +91,7 @@ export class IntegrationService extends RESTService<Integration, Integrations> {
   }
 
   // design time function
+  // @ts-ignore
```
Comment (gashcrumb) [inline app/ui/src/app/store/integration/integration.service.ts]: Good call, I like having an example response customizer function :-)

File: app/ui/src/app/connections/common/configuration/configuration.service.ts (hunk 1 above)
Comment (riccardo-forina) [reply app/ui/src/app/connections/common/configuration/configuration.service.ts]: 🤦‍♂️ I didn't notice that!

**Pull request merged**